src/codewhisperer/service/transformByQHandler.ts

```diff
@@ -140,11 +139,18 @@ export async function validateProjectSelection(project: vscode.QuickPickItem) {
     if (buildFile.length < 1) {
         const buildType = await checkIfGradle(projectPath!)
         vscode.window.showErrorMessage(CodeWhispererConstants.noPomXmlFoundMessage, { modal: true })
+        if (buildType === 'Gradle') {
+            telemetry.codeTransform_isDoubleClickedToTriggerInvalidProject.emit({
+                codeTransformSessionId: codeTransformTelemetryState.getSessionId(),
+                codeTransformPreValidationError: 'NonMavenProject',
+                result: MetadataResult.Fail,
+                reason: buildType,
+            })
+        }
         telemetry.codeTransform_isDoubleClickedToTriggerInvalidProject.emit({
             codeTransformSessionId: codeTransformTelemetryState.getSessionId(),
-            codeTransformPreValidationError: 'Only Maven projects supported' as CodeTransformPreValidationError,
+            codeTransformPreValidationError: 'NoPom',
```

Comment:
could save some repeated code
```suggestion
            codeTransformPreValidationError: buildType === 'Gradle' ? NonMavenProject : 'NoPom',
```